tests/unit/base/trained_model_serialization_unittest.cc.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+
+def read_defined_guards(config_file):
+    with open(config_file) as f:
+        config = f.read().lower()
+        return re.findall('#define (\w+)', config)
+
+
+def is_guarded(include, defined_guards):
+    with open(include) as header:
+        guards = re.findall('#ifdef (\w+)', header.read().lower())
+        return any([g not in defined_guards for g in guards])
+
+
+def is_pure_virtual(name, tags):
+    return any([name + '\timplementation:pure virtual' in tag for tag in tags])
+
+
+def is_shogun_class(c):
+    return c[0] == 'C' and c[1].isupper() and 'class' in c
+
+
+def get_shogun_classes(tags):
+    classes = {}
+    # in ctags format it is TAG\tLOCATION\t..\tinherits:CLASS
```

Comment:
yeah much nicer to parse ctag ouput rather than source code.
One question, isnt there a python package that does things like you do here?